### ISSUE TITLE
cat <<EOM check backquote is closed

### ISSUE THREAD

**Issue by jebsolutions** [koalaman/shellcheck#1135]
#### For bugs
- Rule Id (if any, e.g. SC1000): not sure, think a new rule is needed?
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6 and also online version @ shellcheck.net
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [Y ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ Y] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

	cat <<EOMX >test.txt
	# test `ls -l
EOMX
```

#### Here's what shellcheck currently says:
nothing

#### Here's what I wanted or expected to see:

Some warning about unclosed backquote substitution.

E.g. if I run this script I get an error from /bin/sh:
./test2.sh: 7: ./test2.sh: Syntax error: EOF in backquote substitution

The "cat <<somethinggoeshere" block runs anything in `` quotes in a subshell and includes that output into the cat.  

In the above code I am missing the ending back quote...e.g. should be 
```
# test `ls -l`
```

I only hit this one by accident.  ;)  I wasn't trying to run anything in a subshell.  I copy pasted a comment line that accidentally included the single funky quote and and couldn't figure out why my scripts failed but passed shellcheck.

p.s. shellcheck is great.

**Comment by koalaman** [comment]
Wow. This behavior comes from a very early bugfix where parsing would incorrectly fail for here documents containing `$"`, but there was also a unit test verifying this behavior for backticks for some reason.

It might have been because it's not a syntax error in bash (it's just a bad substitution), and Bash 2 even implicitly terminated them. Failing fast is clearly the better behavior though, and now that's what happens.

Thanks for the bug report, and I'm glad you find ShellCheck useful!
